package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,358 @@
+
+    # extract list of terms:
+    some_terms = aux.return_data(["Potential", "Kinetic En.", "Box-X"])
+
+    # extract all terms:
+    all_terms = aux.return_data()
+    # alternatively
+    all_terms = aux.return_data("*")
+
+.. _EDR: https://manual.gromacs.org/current/reference-manual/file-formats.html#edr
+.. _XDR: https://datatracker.ietf.org/doc/html/rfc1014
+.. _pyedr: https://github.com/mdanalysis/panedr
+.. _GROMACS: https://github.com/gromacs/gromacs/blob/main/src/gromacs/fileio/enxio.cpp
+
+
+.. autoclass:: EDRReader
+   :members:
+
+
+"""
+from pathlib import Path
+from . import base
+import pyedr
+import numpy as np
+from typing import Optional, Union, Dict, List
+
+
+class EDRStep(base.AuxStep):
+    """:class:`AuxStep` class for .edr file format.
+
+    Extends the base AuxStep class to allow selection of time and
+    data-of-interest fields (by dictionary key) from the full set of data read
+    each step.
+
+    Parameters
+    ----------
+    time_selector : str, optional
+        Name of the dictionary key that links to the time values (assumed to
+        be in ps). Default value is "Time"
+    data_selector : str | list of str | None, optional
+        List of dictionary keys linking to data of interest in the EDR file to
+        be stored in ``data``. Default value is ``None``.
+    **kwargs
+        Other AuxStep options.
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxStep`
+    """
+
+    def __init__(self, time_selector: str = "Time",
+                 data_selector: Optional[str] = None, **kwargs):
+        super(EDRStep, self).__init__(time_selector=time_selector,
+                                      data_selector=data_selector,
+                                      **kwargs)
+
+    def _select_time(self, key: str) -> np.float64:
+        """'Time' is one of the entries in the dict returned by pyedr.
+        The base AuxStep Class uses the time_selector 'Time' to return the
+        time value of each step."""
+        return self._select_data(key)
+
+    def _select_data(self, key: Union[str, None]) -> np.float64:
+        if key is None:
+            return
+        try:
+            return self._data[key]
+        except KeyError:
+            raise KeyError(f"'{key}' is not a key in the auxdata dictionary. "
+                           "Check the EDRReader.terms attribute")
+
+
+class EDRReader(base.AuxReader):
+    """ Auxiliary reader to read data from a .edr file.
+
+    Default reader for .edr files. All data from the file will be read and
+    stored on initialisation.
+
+    Parameters
+    ----------
+    filename : str
+        Location of the file containing the auxiliary data.
+    **kwargs
+       Other AuxReader options.
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxReader`
+    :func:`~MDAnalysis.coordinates.base.add_auxiliary`
+
+    Note
+    ----
+    The file is assumed to be of a size such that reading and storing the full
```

Comment:
I wonder at what point this stops being true, EDR files can get quite large right? We might want to keep an eye on memory usage in the future.

Reply:
Not excessively large I think, but on systems with less memory and while multiple instances of EDRReader are open it might become problematic? How would we monitor that?

Reply:
Since many reader instances can be associated with a trajectory at the same time, this would probably need to be multi-level: The AuxReaders could get a new method to determine the total size of their arrays, and on the level of coordinates/base, a new attribute like `aux_size` could be created that is automatically updated when a new auxreader instance is associated with the trajectory, and then warn when it becomes large. Does this sound sensible?

Reply:
Rather than an attribute you constantly change, it may make more sense to make this just a method? You can make it an `@property` if you want to expose it "attribute-like"?

Reply:
This is now implemented in such a way that a memory footprint check is run when a new instance of an AuxReader is added